code/controllers/subsystem/ships.dm

```diff
@@ -533,25 +530,28 @@ SUBSYSTEM_DEF(ship)
 	S.ftl_vector = target
 
 /datum/controller/subsystem/ship/proc/process_factions()
-	for(var/datum/star_faction/faction in star_factions)
+	var/list/star_factions = src.star_factions //cache for free performance
```

Comment:
> free performance

oh fuck off the difference is so immeasurable you should be ashamed to even consider this.